### ISSUE TITLE
Nancy.Testing.Browser / System.Uri throws InvalidOperationException for Relative Uris

### ISSUE THREAD

**Issue by jbattermann** [NancyFx/Nancy#1966]
I just stumbled upon a bit unexpected behaviour in Nancy's (.Testing) Browser:

When making calls on a Browser instance with relative string locations, i.e. "/users" everything works fine and as expected, however passing in a System.Uri with UriKind.Relative throws an InvalidOperationException.

Basically the problem is this:

```
var testBrowser = new Nancy.Testing.Browser(TestBootstrapper, with => { with.HttpRequest(); });
var targetPathAsString = "/users";
var targetPathAsRelativeUri = new System.Uri("/users", UriKind.Relative);

// works:
var responseFromStringPath = testBrowser.Post(targetPathAsString, browserContext => browserContext.JsonBody(someModelInstance, mySerializerInstance));

// does not work / throws InvalidOperationException:
var responseFromRelativeUri = testBrowser.Post(targetPathAsRelativeUri, browserContext => browserContext.JsonBody(someModelInstance, mySerializerInstance));
```

The problem seems to originate from https://github.com/NancyFx/Nancy/blob/master/src/Nancy.Testing/Browser.cs#L302 and https://github.com/NancyFx/Nancy/blob/master/src/Nancy.Testing/Browser.cs#L303 .. basically the two requestUrl Properties (.Scheme and .Hostname) cannot be accessed if the Uri Kind is Relative.. as the Uri property getters throw the exception if the Uri instance is not an Absolute one, see http://referencesource.microsoft.com/#system/net/System/URI.cs,1212 ..

Is this a Bug or a "Feature" of Nancy and I am doing it just wrong? 

Cheers,
-J

**Comment by khellang** [comment]
Are you sure that's where it originates from? Do you have a stack trace? Those properties are not accessed from `System.Uri`, but on [`Nancy.Url`](https://github.com/NancyFx/Nancy/blob/6775d18fc5fa71a191ae80ef154594d210399f45/src/Nancy/Url.cs), and those should've been populated in the constructor [further up](https://github.com/NancyFx/Nancy/blob/6775d18fc5fa71a191ae80ef154594d210399f45/src/Nancy.Testing/Browser.cs#L214-L221).

**Comment by khellang** [comment]
OK, I see what's going on here; the method take a `Nancy.Url`, which has an implicit conversion to `System.Uri` and that's [where the exception happens](https://github.com/NancyFx/Nancy/blob/6775d18fc5fa71a191ae80ef154594d210399f45/src/Nancy/Url.cs#L194-L204). Maybe there should be a check for a relative URL in there...

**Comment by khellang** [comment]
I think I have a fix; #1967

**Comment by jbattermann** [comment]
Hah, that was fast! :) Yeah sorry for the initial, slighly misleading info.. the stack trace here is/was particularly short and reading 'implicit' now makes sense and that's where it happened, yes:

```
   at System.Uri.get_Host()
   at Nancy.Url.op_Implicit(Uri uri)
   at Backend.Api.Tests.Modules.Users.UsersCreationTests.DoesNotAllowEmptySignupRequests() in d:\Development\Backend\Backend.Api.Tests\Modules\Users\UsersCreationTests.cs:line 41
```

Sorry for the confusion but thanks a lot for fixing it so quickly.. WHILE I AM TYPING THIS! ;)
